pkg/lockfile/gemfile-matcher.go

```diff
@@ -0,0 +1,236 @@
+package lockfile
+
+import (
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scanner/internal/cachedregexp"
+
+	"github.com/google/osv-scanner/internal/utility/fileposition"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+const gemfileFilename = "Gemfile"
+const gemField = "gem"
+
+// This is used to clean properties from the gem function syntax
+// We are keeping quotes to stay consistent with the version as we can have multiple version qualifiers
+var gemSyntaxRemover = strings.NewReplacer(",", "")
+
+// We use a specific remover to clean quotes when we fetch the name to find it in the index
+var gemNameSyntaxRemover = strings.NewReplacer(",", "", "\"", "", "'", "")
+
+type GemfileMatcher struct{}
+
+type gemInformation struct {
+	name          string
+	blockLine     models.Position
+	blockColumn   models.Position
+	nameLine      models.Position
+	nameColumn    models.Position
+	versionLine   *models.Position
+	versionColumn *models.Position
+}
+
+func (matcher GemfileMatcher) GetSourceFile(lockfile DepFile) (DepFile, error) {
```

Comment:
If the source file it's in the same folder as the lockfile (or in a subfolder) it could be directly opened like this:
```
return lockfile.Open(gemFilename)
```

Reply:
In fact we don't really want to do that because it does not have the non UTF-8 protection, that's why I used the full method